doc/manual/manual.tex

```diff
@@ -9613,6 +9613,87 @@ \subsubsection{The ``Stokes' law'' benchmark}
 \end{figure}
 
 
+
+\subsubsection{Viscosity grooves benchmark}
+\label{sec:viscosity_grooves}
+
+\textit{This benchmark was designed by Dave May and this section was contributed by Cedric Thieulot} 
+
+The domain is a two-dimensional Cartesian box of size $L\times L$.
+The velocity and pressure fields are given by
+\begin{eqnarray}
+u(x,y) &=& x^3 y + x^2 + xy + x \\
+v(x,y) &=& -\frac{3}{2}x^2y^2 - 2xy - \frac{1}{2}y^2 - y \\
+p(x,y) &=& x^2y^2 + xy + 5 + p_0
+\end{eqnarray}
+where $p_0$ is a constant to be determined based on the type of pressure normalisation.
+The viscosity is chosen to be
+\begin{equation}
+\eta(x,y)=-\sin(p)+1+\epsilon = -\sin (x^2y^2 + xy + 5) + 1 + \epsilon 
+\end{equation}
+where $\epsilon$ controls the viscosity contrast. 
+It is easy to verify that the flow is incompressible as the velocity field satisfies $\nabla\cdot \mathbf u = 0$.
+The right hand side term of the Stokes equation is obtained by inserting 
+the expressions for velocity, pressure and viscosity in the momentum conservation equation \cite{fieldstone}.
+The velocity, pressure and right hand side magnitude are shown in Figure~\ref{fig:benchmark-grooves-3x3} 
+for $L=3$ and $\epsilon=0.1$.
+ 
+The $p_0$ constant can be determined by requiring that the pressure is normalised over the 
+volume of the domain:
+\begin{equation}
+\int_\Omega p dV=
+\int_0^L\int_0^L p(x,y) \; dx dy = 
+\int_0^L\int_0^L (x^2y^2+xy+5) dx dy + \int_0^L \int_0^L p_0 \; dxdy =0
+\end{equation}
+It then follows that:
+\begin{equation}
+p_0 =-  \frac{1}{L^2}  \int_0^L\int_0^L (x^2y^2+xy+5) dx dy
+= -\frac{L^4}{9}-\frac{L^2}{4} - 5. 
+\end{equation}
+
+\begin{figure}
+\centering
+\includegraphics[width=0.31\textwidth]{cookbooks/benchmarks/viscosity_grooves/vel3x3}
+\includegraphics[width=0.31\textwidth]{cookbooks/benchmarks/viscosity_grooves/press3x3}
+\includegraphics[width=0.31\textwidth]{cookbooks/benchmarks/viscosity_grooves/rhs3x3}
+\hfill
+\caption{\it Viscosity grooves benchmark: From left to right, velocity field, pressure field, and 
+norm of the right hand side of the momentum equation, for a $3\times 3$ domain 
+with $\epsilon=0.1$.}
+\label{fig:benchmark-grooves-3x3}
+\end{figure}
+
+As seen in Figure~\ref{fig:benchmark-grooves-domains}, the value of $\epsilon$ controls the viscosity field amplitude:
+when the $\sin$ term of the viscosity takes value 1, the viscosity is then equal to $\epsilon$; when the $\sin$ is 
```

Comment:
```suggestion
when the $\sin$ term of the viscosity takes value -1, the viscosity is then equal to $\epsilon$; when the $\sin$ is 
```

Reply:
there is a minus sign in front of the sin in the expression of the viscosity, so that it is equal to eps when -sin(...)+1 = 0 , i.e. sin(...)=1

Reply:
okay, then the "equal to 1" below is wrong.